### ISSUE TITLE
repeated warning `UserWarning: TypedStorage is deprecated`

### ISSUE THREAD

**Issue by hchau630** [pytorch/pytorch#97207]
### 🐛 Describe the bug

I noticed that the warning `UserWarning: TypedStorage is deprecated` that results from calling `Tensor.storage()` is issued repeatedly, even though the documentation about `torch.set_warn_always` says that by default pytorch warnings should only be issued once per process. While modifying my code to call `untyped_storage()` instead of `storage()` could get rid of the excessive warnings, I would prefer not to since I want to keep my code compatible with pre-2.0.0 versions for now, and `untyped_storage()` is only available >=2.0.0.

Example:
```
import torch

torch.tensor([1,2,3]).storage()
torch.tensor([1,2,3]).storage()
torch.tensor([1,2,3]).storage()
```

Output:
```
/path_to_file/test_warning.py:3: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  torch.tensor([1,2,3]).storage()
/path_to_file/test_warning.py:4: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  torch.tensor([1,2,3]).storage()
/path_to_file/test_warning.py:5: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  torch.tensor([1,2,3]).storage()
```

### Versions

```
Collecting environment information...
PyTorch version: 2.0.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.2.1 (arm64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.28)
CMake version: version 3.26.0
Libc version: N/A

Python version: 3.9.16 (main, Mar  1 2023, 12:19:04)  [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-13.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.24.2
[pip3] torch==2.0.0
[conda] numpy                     1.24.2                   pypi_0    pypi
[conda] torch                     2.0.0                    pypi_0    pypi
```

cc @albanD

**Comment by albanD** [comment]
cc @kurtamohler could you fix that?

As a temporary workaround, you can silent this warning with regular python warning filter: `python -W ignore::UserWarning: foo.py`

**Comment by kurtamohler** [comment]
The complication is that the warning message contains the line of code that triggered the warning, with `warnings.warn(..., stacklevel=...)`:

https://github.com/pytorch/pytorch/blob/0b094ca37f6ac0d3c1899653ea069dbaaf33176b/torch/storage.py#L365-L372

So one option is to remove the `stacklevel` arg. But having the stack in the message is helpful for finding where the warning came from (https://github.com/pytorch/pytorch/pull/89867)

Another option is to only emit the warning the first time `_warn_typed_storage_removal()` gets called. For testing purposes, we could have a flag to make it always warn. Does that seem reasonable?

cc @ezyang

**Comment by ezyang** [comment]
Yeah, let's just have this warn only once and never again. If you're trying to scrub these, you can comment it out

**Comment by lorinczszabolcs** [comment]
Hi! Was this issue already solved? I tried to figure it out based on the PRs, but it seems none of them have been merged / released. Is there any suggested way of silencing it? Thanks!

**Comment by kurtamohler** [comment]
@lorinczszabolcs, #97379 has been merged to fix this issue, but it hasn't been included in a new release yet

Until its released, you could use the workaround mentioned here: https://github.com/pytorch/pytorch/issues/97207#issuecomment-1478692625

But that will silence all `UserWarning`s. If you only want to silence the TypedStorage deprecation warnings, you'd have to do something like this:

```python
import warnings
warnings.filterwarnings('ignore', category=UserWarning, message='TypedStorage is deprecated')
```

**Comment by kurtamohler** [comment]
@RodolpheCalvet, it sounds like you're experiencing a crash related to UntypedStorage/TypedStorage. Could you please open a new issue and cc me on it? https://github.com/pytorch/pytorch/issues/new?assignees=&labels=&template=bug-report.yml
